MapboxMobileEventsTests/MMEEventTests.mm

```diff
@@ -129,6 +135,103 @@
+            NSError *error = nil;
+            MMEEvent *invalid = [MMEEvent eventWithAttributes:MMEEvent.nilAttributes error:&error];
+
+            invalid.attributes should be_nil;
+            error should be_nil;
+        });
+
+        it(@"should not init without MMEEventKeyEvent", ^{
+            NSError *error = nil;
+            MMEEvent *invalid = [MMEEvent eventWithAttributes:@{@"foo":@"bar"} error:&error];
+
+            invalid should be_nil;
+            error should_not be_nil;
+            error.code should equal(MMEErrorEventInit);
+        });
+
+        it(@"should contain an exceptional dictionary", ^{
+            NSError *error = nil;
+            MMEEvent *exceptional = [MMEEvent eventWithAttributes:[MMEExceptionalDictionary dictionaryWithDictionary:@{@"foo":@"bar"}] error:&error];
+
+            exceptional should be_nil;
+            error should_not be_nil;
+            error.code should equal(MMEErrorEventInit);
+        });
+    });
+
+    context(@"eventWithAttributes:", ^{
+        it(@"should return nil with invalid attributes:", ^{
+            MMEEvent *attributed = [MMEEvent eventWithAttributes:@{@"invalid":@"attributes"}];
+
+            attributed should be_nil;
+        });
+    });
+
+    context(@"carplayEvent", ^{
+        it(@"should create a carplay event", ^{
+            MMEEvent *carplay = [MMEEvent carplayEventWithName:MMEventTypeNavigationCarplayConnect attributes:testAttrs];
+
+            carplay should_not be_nil;
+        });
+    });
+
+    context(@"isEqualToEvent:", ^{
+
+        it(@"should not be true for two consective events with the same attributes, or for nil", ^{
+            NSDictionary *eventAttributes = @{MMEEventKeyEvent: @"test.event"};
+            MMEEvent *firstEvent = [MMEEvent eventWithAttributes:eventAttributes];
+            MMEEvent *secondEvent = [MMEEvent eventWithAttributes:eventAttributes];
+
+            [firstEvent isEqual:secondEvent] should_not be_truthy;
+            [firstEvent isEqual:nil] should_not be_truthy;
+            [firstEvent isEqual:firstEvent] should be_truthy;
+            [firstEvent isEqual:eventAttributes] should_not be_truthy;
+        });
+    });
+
+    context(@"hash:", ^{
+        it(@"should compute a non-0 hash", ^{
+            MMEEvent *hashEvent = [MMEEvent eventWithAttributes:@{MMEEventKeyEvent: @"test.event"}];
+            NSUInteger hashcode = hashEvent.hash;
+
+            hashcode should_not equal(0);
+        });
+    });
+
+    context(@"NSCopying", ^{
+        it(@"should create an isEqual: but not identity copy", ^{
+            MMEEvent *original = [MMEEvent eventWithAttributes:@{MMEEventKeyEvent: @"test.event"}];
+            MMEEvent *duplicate = original.copy;
+
+            [original isEqual:duplicate] should be_truthy;
+            (original == duplicate) should_not be_truthy;
+        });
+    });
+
+    context(@"NSSecureCoding", ^{
+        it(@"should create an isEqual: but not identity copy", ^{
```

Comment:
I don't see a difference between this test and the one above. Seems like we could split this up into a couple of `it` blocks and a `beforeEach` as well

Reply:
spooky copy-pasta